plugins/outputs/zabbix/zabbix.go

```diff
@@ -0,0 +1,199 @@
+	m := &ZabbixMetric{Host: host, Key: key, Value: value}
+	// use current time, if `clock` is not specified
+	if m.Clock = time.Now().Unix(); len(clock) > 0 {
+		m.Clock = int64(clock[0])
+	}
+	return m
+}
+
+// ZabbixPacket class.
+type ZabbixPacket struct {
+	Request string    `json:"request"`
+	Data    []*ZabbixMetric `json:"data"`
+	Clock   int64     `json:"clock"`
+}
+
+// ZabbixPacket class cunstructor.
+func NewZabbixPacket(data []*ZabbixMetric, clock ...int64) *ZabbixPacket {
+	p := &ZabbixPacket{Request: `sender data`, Data: data}
+	// use current time, if `clock` is not specified
+	if p.Clock = time.Now().Unix(); len(clock) > 0 {
+		p.Clock = int64(clock[0])
+	}
+	return p
+}
+
+// DataLen ZabbixPacket class method, return 8 bytes with packet length in little endian order.
+func (p *ZabbixPacket) DataLen() []byte {
+	dataLen := make([]byte, 8)
+	JSONData, _ := json.Marshal(p)
+	binary.LittleEndian.PutUint32(dataLen, uint32(len(JSONData)))
+	return dataLen
+}
+
+type Zabbix struct {
+	Host    string
+	Port    int
+	Hosttag string
+
+	Debug bool
+}
+
+var sampleConfig = `
+	# Address of zabbix host
+	host = "zabbix.example.com"
+	# Port of the Zabbix server
+	port = 10051
+	# Which tag will be used for measurement hostname
+	hosttag = "host"
+`
+
+func (z *Zabbix) Connect() error {
+	// Test connection to Zabbix server
+	// format: hostname:port
+	uri := fmt.Sprintf("%s:%d", z.Host, z.Port)
+	tcpAddr, err := net.ResolveTCPAddr("tcp", uri)
+	if err != nil {
+		return fmt.Errorf("Zabbix: TCP address cannot be resolved")
+	}
+
+	connection, err := net.DialTCP("tcp", nil, tcpAddr)
+	if err != nil {
+		return fmt.Errorf("Zabbix: Connection to Zabbix server failed")
+	}
+	defer connection.Close()
+	return nil
+}
+
+func (z *Zabbix) Write(metrics []telegraf.Metric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+	// Send Data to Zabbix server
+	uri := fmt.Sprintf("%s:%d", z.Host, z.Port)
+	tcpAddr, err := net.ResolveTCPAddr("tcp", uri)
+	if err != nil {
+		return fmt.Errorf("Zabbix: TCP address cannot be resolved")
+	}
+
+	connection, err := net.DialTCP("tcp", nil, tcpAddr)
+	if err != nil {
+		return fmt.Errorf("Zabbix: Connection to Zabbix server failed")
+	}
+	defer connection.Close()
+
+	var zabbixMetrics []*ZabbixMetric
+	var hostname string
+	for _, m := range metrics {
+		if h, ok := m.Tags()[z.Hosttag]; !ok {
+			if h, ok := m.Tags()["host"]; !ok {
+				h, err := os.Hostname()
+				if err != nil {
+					return fmt.Errorf("Cannot get os.Hostname()")
+				}
+				hostname = h
+			} else {
+				hostname = h
+			}
+		} else {
+			hostname = h
+		}
+		for fieldName, value := range m.Fields() {
+			metricValue, buildError := buildValue(value)
+			if buildError != nil {
+				fmt.Printf("Zabbix: %s\n", buildError.Error())
+			}
+			if z.Debug {
+				fmt.Printf("%s, %s, %s\n", hostname, fieldName, metricValue)
+			}
+
+			zabbixMetrics = append(zabbixMetrics, NewZabbixMetric(hostname, fieldName, metricValue))
```

Comment:
Otherwise your key can just be something like "available" without any context (in 0.10.x, the `mem` plugin has metric name `mem` and field name `available`, for example)